### ISSUE TITLE
Poison $status and proc_get_last_status after executing background jobs

### ISSUE THREAD

**Issue by ridiculousfish** [fish-shell/fish-shell#1789]
Cloned from #1136 

If a background job is executed, the `$status` variable should generate an error if used, until the next non-background job is used. `proc_get_last_status` should also fail.

**Comment by mqudsi** [comment]
I would argue that background jobs shouldn't touch `$status` and the last interactive command's `$status` should be retained.

**Comment by faho** [comment]
Yeah, and the background status should be reachable via `wait` or something.
